src/merge_sort.py

```diff
@@ -0,0 +1,72 @@
+"""Merge sort a list."""
+import time
+from random import randint
+
+
+def merge_sort(list_in):
+    """Sort a list using the merge sort method."""
+    if len(list_in) > 1:
+        half = len(list_in) // 2
+        list_a = list_in[0:half]
+        list_b = list_in[half:]
+        left = merge_sort(list_a)
+        right = merge_sort(list_b)
+        output = []
+        l_idx = 0
+        r_idx = 0
+        while l_idx < len(left) and r_idx < len(right):
+            if left[l_idx] < right[r_idx]:
+                output.append(left[l_idx])
+                l_idx = l_idx + 1
+            else:
+                output.append(right[r_idx])
+                r_idx = r_idx + 1
+        while l_idx < len(left):
+            output.append(left[l_idx])
+            l_idx = l_idx + 1
```

Comment:
This and the following statements could be two-line if statements that are less obvious iterative processes:
```python
if l_idx < len(left) - 1:
    output.extend(left[l_idx:])
el r_idx < len(right) - 1:
    output.extend(right[r_idx:])
```